### PR TITLE
Date humanizer and admin panel improvements

### PR DESCRIPTION
**Pull request by mbrizic**

### CODE REVIEW

File: server/admin/admin.controller.ts [hunk 1]
```diff
@@ -7,17 +7,25 @@ import UserService from '../users/users.service';
...
+	private authService = new AuthService()
+	private userService = new UserService()
+	private eventsService = new EventsService()
 
 	public getStatsPage = async (req: RequestWithPotentialUser, res: Response, next: NextFunction) => {
-		const events = await this.eventsService.getAllEvents(req.user);
-		const users = await this.userService.getAll();
-		const pageviews = getPageViewsPerDate();
+		const events = await this.eventsService.getAllEvents(req.user)
+		const numberOfPositiveVotes = await this.eventsService.getEventRatingsCount({ votedPositively: true })
```
[the remark this answers is not in the document]
Comment (mbrizic) [reply server/admin/admin.controller.ts:20]: As this is used only on admin panel for now, I didn't worry much about it, but eventually we can refactor it into some `adminService` which would hold such logic and do the parsing there. 

`entitiesToInclude` could be a valid performance issue though. I've written it down to research how we can do custom querying through ORM to only select fields we need and strip unneeded data.

File: server/core/analytics-event.service.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+export function getAnalyticsEvents() {
+    return analyticsEvents
+}
+
+export function clearAnalyticsEvents() {
+    analyticsEvents = []
+}
+
+export type EventType = "UserCreated" | "UserDeleted" | "UserVoted" | "UserRemovedVote"
```
[the remark this answers is not in the document]
Comment (mbrizic) [reply server/core/analytics-event.service.ts:23]: Those are just Typescript string literal types which basically serve as lightweight enums:
https://www.typescriptlang.org/docs/handbook/literal-types.html

So I'd like to keep them just for type safety purposes, but we can refactor it later if needed.

File: server/core/date.service.ts [hunk 3]
```diff
@@ -13,11 +13,37 @@ export function now() {
...
+                return `In exactly ${hoursRemaining} hours`
+            }
+        } else if (timeDiff > minuteInMs) {
+            const diffInMinutes = Math.floor(timeDiff / 1000 / 60);
+            return `In ${diffInMinutes} minutes`;
+        } else {
+            return "Sometime in the future";
+        }
     }
+
```
[the remark this answers is not in the document]
Comment (mbrizic) [reply server/core/date.service.ts:46]: Changed (a lot).

File: database/models/event_rating.ts [hunk 4]
```diff
@@ -4,21 +4,22 @@ import type { Events, EventsId } from './events';
 import type { Users, UsersId } from './users';
 
 export interface EventRatingAttributes {
-  id?: number;
```
Comment (mbrizic) [inline database/models/event_rating.ts:7]: Turns out I'm getting those changes as well, so I've committed them now. 

Must be caused by the changes here (`sequelize-typescript` 1 -> 2 major version update):
https://github.com/mbrizic/ratethegame/pull/21/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R29

File: database/scripts/0002-populate-data.sql [hunk 5]
```diff
@@ -19,4 +19,9 @@ INSERT INTO public.event_rating(would_recommend, created_by, event_id) VALUES
     (true, 2, 1),
     (false, 1, 2),
     (false, 2, 2),
-    (true, 2, 5);
\ No newline at end of file
+    (true, 2, 5);
+
+-- Postgres requires us to reset autoincrement counters manually if we're inserting data by explicitly setting IDs
```
Comment (mbrizic) [inline database/scripts/0002-populate-data.sql:24]: @jurebb This fixes those intermittent "Validation error" issues:

https://til.codes/dont-forget-to-update-the-sequence-in-postgresql-after-a-copy-command/

File: database/models/event_rating.ts (hunk 4 above)
Comment (jurebb) [reply database/models/event_rating.ts:7]: That''s probably it then! I'm on version `2.1.1`

File: database/scripts/0002-populate-data.sql (hunk 5 above)
Comment (jurebb) [reply database/scripts/0002-populate-data.sql:24]: Awesome, great find :D

**Pull request merged**